### DISCUSSION TITLE
I need some help with getting a full list of users playlists!

### DISCUSSION THREAD

**Discussion by HeWhoHoldsAKey** [spotify-web-api-java/spotify-web-api-java#294] [Q&A] [answered]
Hey so I'm currently working on a playlist managing project for my own use and I was going along great until I ran into the issue that I can only gather 20 playlists from a user at a time. This was an issue for me because I have 36 playlists in total. I don't know if there is a workaround to get past the barriers or if there is something wrong with my code.
```Java
import java.io.IOException;

import org.apache.hc.core5.http.ParseException;

import com.spm.main.SpotifyHdlr;
import se.michaelthelin.spotify.SpotifyApi;
import se.michaelthelin.spotify.exceptions.SpotifyWebApiException;
import se.michaelthelin.spotify.requests.data.playlists.GetListOfCurrentUsersPlaylistsRequest;
import se.michaelthelin.spotify.requests.data.users_profile.GetCurrentUsersProfileRequest;
import se.michaelthelin.spotify.model_objects.specification.Paging;
import se.michaelthelin.spotify.model_objects.specification.PlaylistSimplified;
import se.michaelthelin.spotify.model_objects.specification.User;

public class UserObj {
	SpotifyApi sp = SpotifyHdlr.getSAPI();
	static User user;
	static Paging<PlaylistSimplified> playlists;
	GetCurrentUsersProfileRequest currentUserProfileRequest = sp.getCurrentUsersProfile().build();
	GetListOfCurrentUsersPlaylistsRequest currentUsersPlaylistRequest = sp.getListOfCurrentUsersPlaylists().build();
	PlaylistSimplified[] playlistsList;
	public UserObj() {
		try {
			user = currentUserProfileRequest.execute();
			playlists = currentUsersPlaylistRequest.execute();
			playlistsList = playlists.getItems();
			
		} catch (ParseException | SpotifyWebApiException | IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}

	}

	public String getName() {
		return user.getDisplayName();
	}
	
	public Paging<PlaylistSimplified> getUsersPlaylists() {
		return playlists;
	}
	
	public PlaylistSimplified[] getUsersPlaylistsList() {
		return playlistsList;
	}
}
```
I know this probably isn't the most efficient code but I'm still learning and taking it one step at a time!
If yall know of anything I can do to help with this issue id love3 to hear about it and if you want more references to my code I do have a repository that's private at the moment but I can make it public to look through(Or of course I can send it here just giving out options)!

**Comment by HeWhoHoldsAKey** [comment]
Update: I used the .limit() before the builder and set the limit to 50 however, If I were to have over 50 playlists this would still be an issue. So for further and future use and development is there a way to get obtain a full list without limit, or maybe using an offset to get the user's total playlist to count? I'm not sure if I worded it correctly but ill do some work and see what I can find and keep yall updated unless yall have already found a way.

Old code:
```
GetListOfCurrentUsersPlaylistsRequest currentUsersPlaylistRequest = sp.getListOfCurrentUsersPlaylists().build();
```

New code:
```
GetListOfCurrentUsersPlaylistsRequest currentUsersPlaylistRequest = sp.getListOfCurrentUsersPlaylists().limit(50).build();
```

**Comment by dargmuesli** [comment, marked as answer]
The [`GetListOfCurrentUsersPlaylistsRequest`](https://spotify-web-api-java.github.io/spotify-web-api-java/se/michaelthelin/spotify/requests/data/playlists/GetListOfCurrentUsersPlaylistsRequest.html) returns a [`Paging`](https://spotify-web-api-java.github.io/spotify-web-api-java/se/michaelthelin/spotify/model_objects/specification/Paging.html)<[`PlaylistSimplified`](https://spotify-web-api-java.github.io/spotify-web-api-java/se/michaelthelin/spotify/model_objects/specification/PlaylistSimplified.html)>. So you'll need to request he next page(s), e.g. as shown here: https://github.com/spotify-web-api-java/spotify-web-api-java/issues/234#issuecomment-922890827

**Reply by HeWhoHoldsAKey** [reply]
That seems like it would work great! Thanks for the help!
